optaplanner-core/src/test/java/org/optaplanner/core/impl/solver/DefaultSolverTest.java

```diff
@@ -111,4 +115,27 @@ public void solveStopsWhenPartiallyInitialized() {
         assertSame(solution, solver.getBestSolution());
     }
 
+    @Test(timeout = 600_000)
+    public void solveThrowsExceptionWhenZeroEntity() {
+        SolverFactory<TestdataChainedSolution> solverFactory = PlannerTestUtils.buildSolverFactory(
+                TestdataChainedSolution.class, TestdataChainedEntity.class);
+
+        LocalSearchPhaseConfig phaseConfig = new LocalSearchPhaseConfig();
+        phaseConfig.setTerminationConfig(new TerminationConfig().withStepCountLimit(1));
+        solverFactory.getSolverConfig().setPhaseConfigList(Collections.singletonList(phaseConfig));
+
+        Solver<TestdataChainedSolution> solver = solverFactory.buildSolver();
+
+        TestdataChainedSolution solution = new TestdataChainedSolution("1");
+        solution.setChainedEntityList(Collections.EMPTY_LIST);
+        solution.setChainedAnchorList(Collections.singletonList(new TestdataChainedAnchor("4")));
+
+        try {
+            solver.solve(solution);
```

Comment:
If it doesn't throw an exception, this test should fail.
One way to do that is to add a 
            fail("There was no RuntimeException thrown.");
after this line. There are other tests that do that, see PartitionQueueTest.

Besides that, this is a great PR!

Reply:
Thanks! Fixed that.